extensions/Elasticsearch/Elasticsearch/ElasticsearchMemory.cs

```diff
@@ -218,15 +218,20 @@ public async Task<string> UpsertAsync(
         Embedding embedding = await this._embeddingGenerator.GenerateEmbeddingAsync(text, cancellationToken).ConfigureAwait(false);
         var coll = embedding.Data.ToArray();
 
+        //adjust min score for cosine similarity
+        //https://www.elastic.co/guide/en/elasticsearch/reference/current/knn-search.html#knn-similarity-search
+        float adjustedMinSimilarityScore = (float)(2 * minRelevance - 1);
```

Comment:
This formula transforms from score to cosine similarity, however internally ES works with score.

I believe this should be the inverse, e.g. `float adjustedMinSimilarityScore = (float)((minRelevance + 1) / 2)`

By the way, the formula `(2 * minRelevance - 1)` should be used in line 253, replacing `hit.Score` with the actual cosine similarity. The value returned is incorrect and doesn't match the similarity returned by other memory connectors.